pysnmp/proto/secmod/eso/priv/des3.py

```diff
@@ -51,17 +52,19 @@ def hashPassphrase(self, authProtocol, privKey):
                 'Unknown auth protocol %s' % (authProtocol,)
             )
 
-    # 2.1
+    #key localization as per https://tools.ietf.org/html/draft-reeder-snmpv3-usm-3desede-00
     def localizeKey(self, authProtocol, privKey, snmpEngineID):
         if authProtocol == hmacmd5.HmacMd5.serviceID:
             localPrivKey = localkey.localizeKeyMD5(privKey, snmpEngineID)
-            for count in range(1, int(ceil(self.keySize * 1.0 / len(localPrivKey)))):
-                # noinspection PyDeprecation,PyCallingNonCallable
-                localPrivKey += localkey.localizeKeyMD5(localPrivKey, snmpEngineID)
+            #now extend this key if too short by repeating steps that includes the hashPassphrase step
+            while (len(localPrivKey) < self.keySize):
+                newKey = hashPassphraseMD5(localPrivKey)
```

Comment:
So, on entry to `localizeKey`, the` privKey` is not the complete private key yet, it is missing the `P2K(K1,<engine_id>)` step. So the call to:
  `localPrivateKey = localizeKeyMD5/SHA(privKey, snmpEngineId)`
completes the private key, and this is stored in `localPrivateKey`.  

At this point, the key is too short, so then enter the loop where both steps of `hashPassphraseMD5/SHA` and `localizeKeyMD5/SHA` are done until the key is long enough (this is as per the draft standard as you noted).

Then the private key of the right length is returned from the `localPrivKey` variable.